### PR TITLE
Fix $PATH.remove

### PR DESCRIPTION
**Pull request by msoedov**
#2468

### CODE REVIEW

**Comment by astronouth7303** [comment]
1. Can you add a news item for this, per http://xon.sh/devguide.html#changelog ?
2. Can you add a test for this, at least checking basic functionality?

File: tests/test_tools.py [hunk 1]
```diff
@@ -4,33 +4,36 @@
...
-    to_dynamic_cwd_tuple, to_logfile_opt, pathsep_to_set, set_to_pathsep,
-    is_string_seq, pathsep_to_seq, seq_to_pathsep, is_nonstring_seq_of_strings,
-    pathsep_to_upper_seq, seq_to_upper_pathsep, expandvars, is_int_as_str, is_slice_as_str,
-    ensure_timestamp, get_portions, is_balanced, subexpr_before_unbalanced,
-    swap_values, get_logical_line, replace_logical_line, check_quotes, deprecated,
-    )
 from xonsh.environ import Env
-
-from tools import skip_if_on_windows, skip_if_on_unix
+from xonsh.lexer import Lexer
```
Comment (astronouth7303) [inline tests/test_tools.py]: Why the imports reorg?

Comment (msoedov) [inline tests/test_tools.py]: Sorry about that,  it was auto sorted by `isort `

**Comment by msoedov** [comment]
Fixed

**Comment by msoedov** [comment]
One of the recently added unit test is failing with py3.5. I need to investigate this

File: xonsh/tools.py [hunk 2]
```diff
@@ -23,10 +23,10 @@
 import contextlib
 import ctypes
 import datetime
-from distutils.version import LooseVersion
 import functools
 import glob
 import itertools
+import operator
```
Comment (gforsyth) [inline xonsh/tools.py]: Hey @msoedov -- I'd prefer if we didn't touch lines unrelated to actual issues just to alphabetize or similar, it makes git blame/praise less useful for tracking down issues.

File: xonsh/tools.py [hunk 3]
```diff
@@ -35,15 +35,15 @@
 import threading
 import traceback
 import warnings
-import operator
+from distutils.version import LooseVersion
```
Comment (gforsyth) [inline xonsh/tools.py]: ditto on this change

File: xonsh/tools.py [hunk 4]
```diff
@@ -35,15 +35,15 @@
...
 # dependencies
 from xonsh import __version__
-from xonsh.lazyasd import LazyObject, LazyDict, lazyobject
-from xonsh.platform import (has_prompt_toolkit, scandir, DEFAULT_ENCODING,
-                            ON_LINUX, ON_WINDOWS, PYTHON_VERSION_INFO,
-                            expanduser, os_environ)
+from xonsh.lazyasd import LazyDict, LazyObject, lazyobject
+from xonsh.platform import (DEFAULT_ENCODING, ON_LINUX, ON_WINDOWS,
+                            PYTHON_VERSION_INFO, expanduser,
+                            has_prompt_toolkit, os_environ, scandir)
```
Comment (gforsyth) [inline xonsh/tools.py]: and here and the lines above -- if we aren't adding anything then leave the imports alone

File: tests/test_tools.py [hunk 5]
```diff
@@ -788,6 +788,27 @@ def test_env_path_slice_get_all_except_first_element(inp, exp):
...
+    env_path = EnvPath([])
+    assert len(env_path) == 0
+    env_path.append(path)
+    assert len(env_path) == 1
+
+
+def test_env_path_eq():
+    env_path = EnvPath([pathlib.Path("~/node_modules/.bin")])
+    list_paths = [pathlib.Path("~/node_modules/.bin")]
+    assert env_path == list_paths
```
Comment (gforsyth) [inline tests/test_tools.py]: `pathlib` doesn't expand `~` automatically which might be causing the test failure on this

**Comment by scopatz** [comment]
It seems like there is a test failure here.

**Comment by astronouth7303** [comment]
Indeed there is, but @msoedov hasn't addressed them or @gforsyth's comments.

**Comment by gforsyth** [comment]
Hey @msoedov -- do you want to continue work on this?  I think we're just a few steps away from getting this in if you'd like to pick it back up.  Otherwise we can try to incorporate your work into a separate patch.

Thanks!

**Comment by scopatz** [comment]
Closing since this has been open for more than a year. Please feel free to reopen!

**Pull request closed without merging**